src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,117 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int threshold;
+    private int capacity;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null || table.length == 0) {
+            table = new Node[DEFAULT_CAPACITY];
+        }
+        capacity = table.length;
+        threshold = (int) (capacity * LOAD_FACTOR);
+        if (size >= threshold) {
+            table = resize();
+        }
+        int index = Math.abs(hash(key) % capacity);
```

Comment:
Put getting the index into a separate method. Also in new method if key==null return 0